packages/client/src/user/userReducer.ts

```diff
@@ -404,18 +404,19 @@ export const userFormReducer: LoopReducer<IUserFormState, UserFormAction> = (
     case ROLES_LOADED:
       const { loggedInUserScopes, userRoles } = action.payload
 
-      const roleScopes = (role: string) =>
-        userRoles.find(({ id }) => id === role)?.scopes ?? []
+      const creatableRoleIds =
+        loggedInUserScopes
+          .map((rawScope) => parseScope(rawScope))
+          .find((parsedScope) => parsedScope?.type === 'user.create')?.options
+          ?.role ?? []
 
-      const canCreateOrUpdateAll = [
-        SCOPES.USER_CREATE,
-        SCOPES.USER_UPDATE
-      ].some((scope) => loggedInUserScopes.includes(scope))
+      const editableRoleIds =
+        loggedInUserScopes
+          .map((rawScope) => parseScope(rawScope))
+          .find((parsedScope) => parsedScope?.type === 'user.edit')?.options
+          ?.role ?? []
 
-      const canCreateOrUpdateJurisdiction = [
-        SCOPES.USER_CREATE_MY_JURISDICTION,
-        SCOPES.USER_UPDATE_MY_JURISDICTION
-      ].some((scope) => loggedInUserScopes.includes(scope))
+      const allowedRoleIds = [...creatableRoleIds, ...editableRoleIds]
```

Comment:
@Zangetsu101 this made me think, can we create helper

```ts
function findScope<Scope extends ConfigurableScopes>(rawScopes: RawScope[], scope: Scope): Scope | undefined
```

This code could then become 

```ts
      const creatableRoleIds =
        findScope(loggedInUserScopes, CreateUserScope)?.options.role ?? []

      const editableRoleIds =
        findScope(loggedInUserScopes, EditUserScope)?.options.role ?? []
```

Reply:
I had something like this in mind: `findScope(loggedInUserScopes, 'user.create')`
So which one would be preferable?

Reply:
I'm sure someone from Events v2 team would want to do it using an enum so `Scopes.____` instead of a string literal but otherwise I do agree

Reply:
Using this syntax, it's easier to narrow down the type 😅